src/components/movie-card/movie-card.jsx

```diff
@@ -1,39 +1,137 @@
-import { useState, useEffect } from "react";
+import React from "react";
+import React, { useState, useEffect } from "react";
 import PropTypes from "prop-types";
-import {Button, Card} from "react-bootstrap";
-//import "./movie-card.scss";
-export const MovieCard = ({ movie, onMovieClick }) => {
-    return (
+import { Button, Card } from "react-bootstrap";
+import { Link } from "react-router-dom";
+
+export const MovieCard = ({ movie, user, token, setUser }) => {
+  const [favorite, setFavorite] = useState(user && user.FavoriteMovies && user.FavoriteMovies.includes(movie.title));
+
+
+  const storedToken = localStorage.getItem("token");
+  const storedUser = JSON.parse(localStorage.getItem("user"));
+  //const [user, setUser] = useState(storedUser ? storedUser : null);
+  //const [token, setToken] = useState(storedToken ? storedToken : null);
+
+  //const [addTitle, setAddTitle] = useState("");
+  //const [delTitle, setDelTitle] = useState("");
+
+  const [favoriteMovies, setfavoriteMovies] = useState(false);
+
+
+  //useEffect(() => {
+    //setfavoriteMovies(user.favoritemovie && user.favoritemovie.includes(movie.title));
+
+  //}, [user, movie.title]);
+
+  const addToFavorites = () => {
+    fetch(
+      `https://movie-api-7p14.onrender.com/users/${user.userName}/movies/${movie.title}`,
```

Comment:
You have called `user.userName` But what you have in your localStorage is `user.Username`.